.github/workflows/doc.yml

```diff
@@ -53,6 +53,27 @@ jobs:
     - name: 'Build HTML documentation'
       run: make -C Doc/ SPHINXOPTS="-q" SPHINXERRORHANDLING="-W --keep-going" html
 
+    # Add pull request annotations for Sphinx nitpicks (missing references)
+    - name: Get list of changed files
+      id: changed_files
+      uses: Ana06/get-changed-files@v2.2.0
+    - name: 'Build changed files with nitpicks'
+      continue-on-error: true
+      run: |
+        # Mark files the pull request modified
+        touch ${{ steps.changed_files.outputs.added_modified }}
+        # Build docs with the '-n' (nit-picky) option, convert warnings
+        make -C Doc/ PYTHON=../python SPHINXOPTS="-q -n --keep-going" html 2>&1 |
+            python Doc/tools/warnings-to-gh-actions.py
+
+    # Ensure some files always pass Sphinx nitpicks (missing references)
+    - name: 'Build others with nitpicks'
+      run: |
+        # Mark files that must pass nit-picky
+        touch Doc/whatsnew/3.12.rst
```

Comment:
Especially since this list will (hopefully) get rather long (though using globs will help reduce it), IMO it would be cleaner to store the list of file/directory globs in a separate file (e.g. a text file in the `Doc/` directory), one path/glob per line, and then pipe in and iterate over the lines here (`touch`-ing each one, of course).

That would make it easier and more visible to update without going digging, and would also make it way easier for other tooling, scripts, etc. to use it as well, and if an option is added to upstream Sphinx, we could simply switch to reading it in the `conf.py` instead of having to convert it to a whole other format (which could also create problems with backports).

Reply:
Sounds good, possibly even a bash file that touches them. Anyway, will leave that for a follow-up so we can get this merged and see how it works in action.